t/segment.t

```diff
@@ -0,0 +1,73 @@
+use Test2::V0;
+use lib 't/lib';
+
+use NewFangle::Agent;
+use NewFangle::Agent::SegmentStarter::HTTP::Tiny;
+use NewFangle::Agent::SegmentStarter::LWP::UserAgent;
+use NewFangle::Agent::SegmentStarter::DBI::st;
+
+package Foo { sub foo {} };
+package Bar { sub bar {} };
+package Baz { sub baz {} };
+
+{
+    my $mock = local $NewFangle::Agent::TX = mock {} => add => [
+         start_segment => sub { shift->{segment} = \@_ },
+    ];
+
+    NewFangle::Agent::install_wrappers('Foo'); # uses custom starter
+    Foo->foo(qw/1 2 3/);
+    is $mock->{segment}, ['Foo::foo', [qw/1 2 3/]], 'Foo::foo uses custom starter';
+
+    like dies { NewFangle::Agent::install_wrappers('Bar') },
+        qr/does not implement install/, 'die when install is not implemented';
+
+    NewFangle::Agent::install_wrappers('Baz'); # uses default starter
+    Baz->baz;
+    is $mock->{segment}, ['Baz::baz', ''], 'Baz::baz uses default starter';
+
+}
+
+{
+    my $class = 'NewFangle::Agent::SegmentStarter::HTTP::Tiny';
+    my $mock = local $NewFangle::Agent::TX
+        = mock {} => add => [ start_external_segment => 'rw' ];
+    ok my $sub = $class->install('request'), 'custom starter for HTTP::Tiny::request';
+    HTTP::Tiny->$sub( 'get', 'https://example.com?foo=1' );
+    like $mock->start_external_segment,
+        [qw[https://example.com GET HTTP::Tiny]], 'external segment has correct params';
+}
+
+{
+    my $class = 'NewFangle::Agent::SegmentStarter::LWP::UserAgent';
+    my $mock = local $NewFangle::Agent::TX
+        = mock {} => add => [ start_external_segment => 'rw' ];
+    ok my $sub = $class->install('request'), 'custom starter for LWP::UserAgent::request';
+    my $req = mock {} => add =>
+        [ url => sub {'https://example.com?foo=1'}, method => 'get' ];
+    LWP::UserAgent->$sub($req);
```

Comment:
Will this pass if LWP::UserAgent is not installed?

Reply:
Yes:
```
$ perl -E 'my $sub = sub { say for @_ }; Does::Not::Exist->$sub(qw/foo bar baz/)'
Does::Not::Exist
foo
bar
baz
```

Reply:
It doesn't resolve the symbol but passes the 'class' as first parameter.

Reply:
Is this too magical? :sweat_smile:

Reply:
> Yes:
> 
> ```
> $ perl -E 'my $sub = sub { say for @_ }; Does::Not::Exist->$sub(qw/foo bar baz/)'
> Does::Not::Exist
> foo
> bar
> baz
> ```

It's sugar for `$sub->('Does::Not::Exist', qw/foo bar baz/)`